File: daily-coding-challenge/2025/august/js/sumOfSquares.js
/*

Sum of Squares

Given a positive integer up to 1,000, return the sum of all the integers squared from 1 up to the number.

Tests:
  sumOfSquares(5) should return 55.
  sumOfSquares(10) should return 385.
  sumOfSquares(25) should return 5525.
  sumOfSquares(500) should return 41791750.
  sumOfSquares(1000) should return 333833500.

*/

function sumOfSquares(n) {
  let result = 0;
  for (let i = 1; i <= n; i++) {
    result += i ** 2; // i * i
  }
  return result;
}

console.log(sumOfSquares(5)); // 55
console.log(sumOfSquares(10)); // 385
console.log(sumOfSquares(25)); // 5525
console.log(sumOfSquares(500)); // 41791750
console.log(sumOfSquares(1000)); // 333833500

File: daily-coding-challenge/2025/august/py/sum_of_squares.py
"""

Sum of Squares

Given a positive integer up to 1,000, return the sum of all the integers squared from 1 up to the number.

Tests:
  sum_of_squares(5) should return 55.
  sum_of_squares(10) should return 385.
  sum_of_squares(25) should return 5525.
  sum_of_squares(500) should return 41791750.
  sum_of_squares(1000) should return 333833500.

"""


def sum_of_squares(n):
    total_sum = 0
    for i in range(1, n + 1):
        total_sum += i * i
    return total_sum


print(sum_of_squares(5))  # 55
print(sum_of_squares(10))  # 385
print(sum_of_squares(25))  # 5525
print(sum_of_squares(500))  # 41791750
print(sum_of_squares(1000))  # 333833500
